### PR TITLE
Removes some static global variables

### PR DESCRIPTION
**Pull request by TinyTinni**
Hi,
**Background to my current problem**:
I have a shared library which static links duckdb into it. When the shared library gets unloaded under linux, there is a double free in all? some? of duckdbs global static variables. This happens since 0.9.0. Thanks to sanitizers, I could track down the problem to the some problematic global variables. They are all string types. (It probably has something to do with static init/uninit ordering).

This PR eliminates the usage of the problematic global string variables. Most are const anyway and can be abstracted to a function or were not used anyway.

**Uncertainty**
One uncertain part are the changes in "parquet_types.h" and it's .cpp counterpart. Those files seems to be originally generated, but got also slightly changed over time. The files were introduced with miniparquet, which is not part of duckdb anymore. The files were also slightly modified already. I couldn't find any thrift script in the duckdb repo which let's me generate those files.

The operators do not use a global std::map anymore. This has also the advantage, that the compiler (at least clang+gcc) emits a warning, if the switch/cases do not cover all possibilities defined in the enumeration.

I would be happy about any comments to the PR and/or a merge.

Thank you for taking your time going through it.

### CODE REVIEW

File: src/main/extension/generated_extension_loader.cpp.in [hunk 1]
```diff
@@ -9,10 +9,17 @@ ${EXT_LOADER_BODY}
     return false;
 }
 
-vector<string> linked_extensions = {${EXT_NAME_VECTOR_INITIALIZER}
-};
+const vector<string>& LinkedExtensions(){
+    static vector<string> VEC = {${EXT_NAME_VECTOR_INITIALIZER}
```
Comment (Tishj) [inline src/main/extension/generated_extension_loader.cpp.in:13]: Either the tidy check or the formatter is not going to be happy with VEC as variable name, `vec` is accepted

Comment (TinyTinni) [reply src/main/extension/generated_extension_loader.cpp.in:13]: The formatter pipeline/action complained about `vec` when I was running it. Same with the REGEX variable. 

Or I was doing something wrong.

Would happily change it to lowercase.

Comment (TinyTinni) [reply src/main/extension/generated_extension_loader.cpp.in:13]: I am surprised clang-tidy didn't complain about the global statics but the local ones... Feels flipped.

**Comment by TinyTinni** [comment]
The job "Java JDBC / Java Linux (aarch64)" failed. Looks for me like a setup error. Unlikely it was me, but I cannot rerun the job. 

Can you do it or should I push a dummy commit/amned commit+force push?

**Comment by Mytherin** [comment]
Thanks! LGTM. I agree that it is in general a good idea to avoid static variables with non-trivial destructors.

> One uncertain part are the changes in "parquet_types.h" and it's .cpp counterpart. Those files seems to be originally generated,

Indeed those files are originally from Thrift, but we don't re-generate them anymore so changing them at this point is alright (although still ideally avoided if possible).

**Pull request merged**